File: python/997.Find-the-Town-Judge.py
class Solution:
    def findJudge(self, n: int, trust: List[List[int]]) -> int:
        trust_list = [0] * n

        for j1, j2 in trust:
            trust_list[j1 - 1] -= 1
            trust_list[j2 - 1] += 1

        for i in range(n):
            if trust_list[i] == n - 1:
                return i + 1

        return -1

File: javascript/997.Find-the-Town-Judge.js
/**
 * @param {number} n
 * @param {number[][]} trust
 * @return {number}
 */
var findJudge = function (n, trust) {
  const incomingMap = new Map();
  const outgoingMap = new Map();
  const incomingEdges = n - 1;
  const outgoingEdges = 0;

  for (let i = 1; i <= n; i++) {
    incomingMap.set(i, 0);
    outgoingMap.set(i, 0);
  }

  for (const [j1, j2] of trust) {
    incomingMap.set(j2, incomingMap.get(j2) + 1);
    outgoingMap.set(j1, outgoingMap.get(j1) + 1);
  }

  for (let i = 1; i <= n; i++) {
    if (
      incomingMap.get(i) === incomingEdges &&
      outgoingMap.get(i) === outgoingEdges
    ) {
      return i;
    }
  }

  return -1;
};
